src/ray/rpc/grpc_server.h

```diff
@@ -31,15 +31,15 @@ class GrpcServer {
   /// \param[in] name Name of this server, used for logging and debugging purpose.
   /// \param[in] port The port to bind this server to. If it's 0, a random available port
   ///  will be chosen.
-  GrpcServer(const std::string &name, const uint32_t port)
-      : name_(name), port_(port), is_closed_(false) {}
+  GrpcServer(std::string name, const uint32_t port)
+      : name_(std::move(name)), port_(port), is_closed_(false) {}
 
   /// Construct a gRPC server that listens on unix domain socket.
   ///
   /// \param[in] name Name of this server, used for logging and debugging purpose.
   /// \param[in] unix_socket_path Unix domain socket full path.
-  GrpcServer(const std::string &name, const std::string &unix_socket_path)
-      : GrpcServer(name, 0) {
+  GrpcServer(std::string name, const std::string &unix_socket_path)
+      : GrpcServer(std::move(name), 0) {
```

Comment:
I previously had the same question, but actually this does the right thing and is the suggested style - it can avoid the copy if user calls the constructor with an explicit `std::move`. On the other hand, I agree that for not performance critical code like this, an extra copy is also ok, and can just use `const &`.

https://stackoverflow.com/questions/51705967/advantages-of-pass-by-value-and-stdmove-over-pass-by-reference/51706522

Reply:
This change was actually introduced by my PR. Please see my comment here https://github.com/ray-project/ray/pull/5196#discussion_r303291390